test/integration/cat/test_cat.py

```diff
@@ -6,51 +6,65 @@
 
 from utils import run_command_and_get_output
 
+TEST_CAT_ARGS_CCONFIG = ("args, cat_result, found, not_found")
 
-def test_cat_unset_arg(tt_cmd, tmp_path):
-    # Testing with unset .xlog or .snap file.
-    cmd = [tt_cmd, "cat"]
-    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
-    assert rc == 1
-    assert re.search(r"it is required to specify at least one .xlog or .snap file", output)
-
-
-def test_cat_non_existent_file(tt_cmd, tmp_path):
-    # Testing with non-existent .xlog or .snap file.
-    cmd = [tt_cmd, "cat", "path-to-non-existent-file"]
-    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
-    assert rc == 1
-    assert re.search(r"No such file or directory", output)
-
-
-def test_cat_snap_file(tt_cmd, tmp_path):
-    # Copy the .snap file to the "run" directory.
-    test_app_path = os.path.join(os.path.dirname(__file__), "test_file", "test.snap")
-    shutil.copy(test_app_path, tmp_path)
 
-    # Testing .snap file.
-    cmd = [
-        tt_cmd, "cat", "test.snap", "--show-system",
-        "--space=320", "--space=296", "--from=423", "--to=513"
-        ]
-    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
-    assert rc == 0
-    assert re.search(r"lsn: 423", output)
-    assert re.search(r"lsn: 512", output)
-    assert re.search(r"space_id: 320", output)
-    assert re.search(r"space_id: 296", output)
+def make_test_cat_args_param(
+    args=[],
+    cat_result=0,
+    found={},
+    not_found={},
+):
+    return pytest.param(args, cat_result, found, not_found)
 
 
-def test_cat_xlog_file(tt_cmd, tmp_path):
+@pytest.mark.parametrize(TEST_CAT_ARGS_CCONFIG, [
+    make_test_cat_args_param(
+        # Testing with unset .xlog or .snap file.
+        cat_result=1,
```

Comment:
> return code of the external utility?
Nope, this is result of `tt cat <args>` command execution.

> split the test
Now they divided to `common` part and `timestamp` testing part. I thought that a good idea not to mix them.

Reply:
You're right about the “good idea not to mix them”, that's the reason why it's better to stick with that idea to the end and separate them into successful and unsuccessful ones as well.
So it would be something like:

- common success
- common fail
- timestamp success
- timestamp fail

This separation will make the code simpler, remove unnecessary parameterization at the input and unnecessary flow control _(if-else)_ inside the test function.